.github/workflows/publish.yml

```diff
@@ -23,27 +22,27 @@ jobs:
 
       - name: Install Dependencies
         run: yarn
+      - name: Update package.json version
+        run: |
+          if [ "${{ github.event_name }}" = "release" ]; then
+            TAG_NAME="${{ github.event.release.tag_name }}"
+            VERSION=${TAG_NAME#v}  # Remove 'v' prefix if present
+            npm version $VERSION --no-git-tag-version
+            echo "Updated package.json version to $VERSION"
+          else
+            echo "Skipping version update for manual trigger"
+          fi
+
+      - name: Commit version update
+        if: github.event_name == 'release'
+        run: |
+          git config --local user.email "action@github.com"
+          git config --local user.name "GitHub Action"
+          git add package.json
+          git commit -m "chore: update version to ${{ github.event.release.tag_name }}"
+          git push
```

Comment:
We've been using an action to do these commits, e.g. https://github.com/atomicfi/services/blob/master/.github/workflows/helm-charts-test.yaml#L73-L78